app/models/concerns/second_factor_manager.rb

```diff
@@ -45,6 +45,10 @@ def backup_codes_enabled?
       SiteSetting.enable_local_logins
   end
 
+  def remaining_backup_codes
+    self&.user_second_factors&.backup_codes&.length
```

Comment:
We usually want to do a `.count` instead of `.length` here. `.length` loads all the associations into an array and counts the length of the array which is memory inefficient. `.count` on the other hands converts the SQL query into a count query and returns an integer.